collect_app/src/test/java/org/odk/collect/android/projects/QrCodeProjectCreatorDialogTest.kt

```diff
@@ -128,4 +131,37 @@ class QrCodeProjectCreatorDialogTest {
             Intents.release()
         }
     }
+
+    @Test
+    fun `When QR code is invalid a toast should be displayed`() {
+        val projectCreator = mock<ProjectCreator>()
+
+        CollectHelpers.overrideAppDependencyModule(object : AppDependencyModule() {
+            override fun providesBarcodeViewDecoder(): BarcodeViewDecoder {
+                val barcodeResult = mock<BarcodeResult> {
+                    `when`(it.text).thenReturn("%")
+                }
+
+                return mock {
+                    `when`(it.waitForBarcode(any())).thenReturn(MutableLiveData(barcodeResult))
+                }
+            }
+            override fun providesProjectCreator(
+                projectImporter: ProjectImporter?,
+                projectsRepository: ProjectsRepository?,
+                currentProjectProvider: CurrentProjectProvider?,
+                settingsImporter: SettingsImporter?,
+                context: Context,
+                storagePathProvider: StoragePathProvider
+            ): ProjectCreator {
+                return projectCreator
+            }
+        })
+
+        val scenario = RobolectricHelpers.launchDialogFragmentInContainer(QrCodeProjectCreatorDialog::class.java, R.style.Theme_MaterialComponents)
+        scenario.onFragment {
```

Comment:
Not a big deal, but just to point out that we don't need the assertions to be in `onFragment` here. That's only needed if we have to access the `Fragment` itself. Good to change before merging!

Reply:
Fixed.